public/app/core/components/TimelineChart/utils.ts

```diff
@@ -525,9 +525,9 @@ export function getThresholdItems(fieldConfig: FieldConfig, theme: GrafanaTheme2
   }
 
   const steps = thresholds.steps;
-  const disp = getValueFormat(thresholds.mode === ThresholdsMode.Percentage ? 'percent' : fieldConfig.unit ?? '');
+  const getDisplay = getValueFormat(thresholds.mode === ThresholdsMode.Percentage ? 'percent' : fieldConfig.unit ?? '');
 
-  const fmt = (v: number) => formattedValueToString(disp(v));
+  const format = (value: number) => formattedValueToString(getDisplay(value, fieldConfig.decimals ?? 0));
```

Comment:
do you want to default to 0?  i think undefined will pick the auto decimals when not set

Reply:
I'll test this out locally and see what the behavior is 👍

Reply:
this second arg wasnt really meant for this, but should work :)

have not looked at the code, but why wouldnt the regular displayProcessor respect the decimals specified in the options/fieldconfig without needing this additional override? 🤔